### PR TITLE
Falling into a chasm doesn't delete your body and someone can fish you up

### PR DESCRIPTION
**Pull request by ToasterBiome**
# Document the changes in your pull request

Falling into a chasm as a living mob will instead very kill you and store you in the contents, using a rod after 3 seconds will pull all (now very dead) living mob corpses up for rescue. No outward indication that anything is in the chasm so that there's still a punishment (better have that GPS on!!!)

mining medic gets a collapsible rod in the mining based named "ol' reliable"

Steals partially from https://github.com/tgstation/tgstation/pull/69252

# Changelog

:cl:  ToasterBiome, Jacquerel
rscadd: mining medic gets a 0 fishing power fishing rod for fishing up miner corpses (lol)
tweak: falling in chasms no longer deletes your body
/:cl:

### CODE REVIEW

**Comment by JamieD1** [comment]
JamieSeal

**Comment by JamieD1** [comment]
FIRE AXE CABINET FISHING ROD

**Comment by LazennG** [comment]
@ToasterBiome 
fish yo miner bithc

**Comment by MajManatee** [comment]
i love fishing

File: code/datums/components/chasm.dm [hunk 1]
```diff
@@ -133,11 +134,48 @@
...
+	if(do_after(user, 3 SECONDS, src.parent))
+		if(!rod.wielded)
+			return
+		var/atom/parent = src.parent
+		var/list/fishing_contents = parent.GetAllContents()
+		if(!length(fishing_contents))
+			to_chat(user, span_warning("There's nothing here!"))
+			return
+		var/found = FALSE
+		for(var/mob/M in fishing_contents) //since only mobs can fall in here this really isnt needed but on the off chance something naughty happens..
```
Comment (Moltijoe) [inline code/datums/components/chasm.dm:225]: can't objects fall in too? ie, cut off someone's head and throw it in

File: code/datums/components/chasm.dm [hunk 2]
```diff
@@ -133,11 +134,48 @@
...
 			var/atom/parent = src.parent
-			parent.visible_message(span_boldwarning("[parent] spits out [AM]!"))
+			AM.forceMove(parent)
 			AM.alpha = oldalpha
 			AM.color = oldcolor
 			AM.transform = oldtransform
-			AM.throw_at(get_edge_target_turf(parent,pick(GLOB.alldirs)),rand(1, 10),rand(1, 10))
+			var/mob/living/L = AM
+			L.Paralyze(100)
+			L.adjustBruteLoss(400)
```
Comment (Moltijoe) [inline code/datums/components/chasm.dm]: Memento Mori? They wouldnt die, but I kinda think that sounds funny

Comment (ynot01) [inline code/datums/components/chasm.dm]: Bones should break on a fall like this, adjustBruteLoss does not roll for wounds

**Review by ynot01 — DISMISSED** [review]

File: code/datums/components/chasm.dm (hunk 1 above)
Comment (Moltijoe) [reply code/datums/components/chasm.dm:225]: ah, you mean only mobs are added to the contents, everything else is just deleted

File: code/datums/components/chasm.dm (hunk 2 above)
Comment (MajManatee) [reply code/datums/components/chasm.dm]: give them a compond fracture on both legs

**Comment by bruhlookatthisdood** [comment]
make it have a chance to catch a bubble gum

**Review by cuackles — APPROVED** [review]
only thning i could ask for to change is to add a unique inhand sprite but that isnt even necesary being that you the other ones already use the same inhand and they dont have any notable mechanical diferences that would merit that change so, good sprite 🐟

**Comment by ShoziOffical** [comment]
Does this fishing rod still based on fishing power to hook up miners from chasm? I suppose it should be because mining medic can sit on boat or dragon from lavaland to increase their fishing power

**Comment by warface1234455** [comment]
Good use for boats

**Comment by ToasterBiome** [comment]
> Does this fishing rod still based on fishing power to hook up miners from chasm? I suppose it should be because mining medic can sit on boat or dragon from lavaland to increase their fishing power

No it doesn't

File: code/datums/components/chasm.dm [hunk 3]
```diff
@@ -133,11 +134,48 @@
...
+		if(!rod.wielded)
+			return
+		var/atom/parent = src.parent
+		var/list/fishing_contents = parent.GetAllContents()
+		if(!length(fishing_contents))
+			to_chat(user, span_warning("There's nothing here!"))
+			return
+		var/found = FALSE
+		for(var/mob/M in fishing_contents) //since only mobs can fall in here this really isnt needed but on the off chance something naughty happens..
+			M.forceMove(get_turf(parent))
```
Comment (ThatLing) [inline code/datums/components/chasm.dm]: Wouldn't this make them instantly fall in again?

Comment (ToasterBiome) [reply code/datums/components/chasm.dm]: yeah probably

**Review by patpol4 — APPROVED** [review]
map ok

File: code/datums/components/chasm.dm [hunk 4]
```diff
@@ -86,58 +88,147 @@
...
+ */
+/datum/component/chasm/proc/left_chasm(atom/source, atom/movable/gone)
+	SIGNAL_HANDLER
+	UnregisterSignal(gone, COMSIG_LIVING_REVIVE)
+
+#define CHASM_TRAIT "chasm trait"
+
+/**
+ * Called if something comes back to life inside the pit. Expected sources are badmins and changelings.
+ * Ethereals should take enough damage to be smashed and not revive.
```
Comment (Moltijoe) [inline code/datums/components/chasm.dm]: this comment should probably be removed, our ethereals don't do this

Comment (ToasterBiome) [reply code/datums/components/chasm.dm]: Ethereals don't shatter?

Comment (Moltijoe) [reply code/datums/components/chasm.dm]: This is referring to them self reviving using their crystal core. Ethereals on are server are closer slimes than crystals

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**